<?xml version="1.0" encoding="UTF-8"?>
<pmd xmlns="http://pmd.sourceforge.net/report/2.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pmd.sourceforge.net/report/2.0.0 http://pmd.sourceforge.net/report_2_0_0.xsd" version="6.44.0" timestamp="2023-10-12T18:05:07.039">
<file name="/mnt/c/Users/Valentino/Documents/Universita/Tesi/Tesi/Pacchetto_di_replicabilita/Correttiva/src/CWE-22/good$aa0c2c26dde8930cad924796af7c17a13d236b16$279$RunSQLHelper.java">
<violation beginline="32" endline="32" begincolumn="26" endcolumn="73" rule="ProperLogger" ruleset="Error Prone" package="org.apache.geronimo.console.internaldb" class="RunSQLHelper" variable="VariableDeclaratorId" externalInfoUrl="https://pmd.github.io/pmd-6.44.0/pmd_rules_java_errorprone.html#properlogger" priority="3">
Logger should be defined private static final and have the correct class
</violation>
<violation beginline="61" endline="67" begincolumn="11" endcolumn="9" rule="AvoidCatchingThrowable" ruleset="Error Prone" package="org.apache.geronimo.console.internaldb" class="RunSQLHelper" method="createDB" externalInfoUrl="https://pmd.github.io/pmd-6.44.0/pmd_rules_java_errorprone.html#avoidcatchingthrowable" priority="3">
A catch statement should never catch throwable since it includes errors.
</violation>
<violation beginline="62" endline="62" begincolumn="17" endcolumn="17" rule="AvoidInstanceofChecksInCatchClause" ruleset="Error Prone" package="org.apache.geronimo.console.internaldb" class="RunSQLHelper" method="createDB" externalInfoUrl="https://pmd.github.io/pmd-6.44.0/pmd_rules_java_errorprone.html#avoidinstanceofchecksincatchclause" priority="3">
An instanceof check is being performed on the caught exception.  Create a separate catch clause for this exception type.
</violation>
<violation beginline="144" endline="144" begincolumn="14" endcolumn="24" rule="UselessParentheses" ruleset="Code Style" package="org.apache.geronimo.console.internaldb" class="RunSQLHelper" method="runSQL" externalInfoUrl="https://pmd.github.io/pmd-6.44.0/pmd_rules_java_codestyle.html#uselessparentheses" priority="4">
Useless parentheses.
</violation>
<violation beginline="144" endline="144" begincolumn="31" endcolumn="54" rule="UselessParentheses" ruleset="Code Style" package="org.apache.geronimo.console.internaldb" class="RunSQLHelper" method="runSQL" externalInfoUrl="https://pmd.github.io/pmd-6.44.0/pmd_rules_java_codestyle.html#uselessparentheses" priority="4">
Useless parentheses.
</violation>
<violation beginline="173" endline="174" begincolumn="15" endcolumn="13" rule="EmptyCatchBlock" ruleset="Error Prone" package="org.apache.geronimo.console.internaldb" class="RunSQLHelper" method="runSQL" externalInfoUrl="https://pmd.github.io/pmd-6.44.0/pmd_rules_java_errorprone.html#emptycatchblock" priority="3">
Avoid empty catch blocks
</violation>
<violation beginline="176" endline="178" begincolumn="11" endcolumn="9" rule="AvoidCatchingThrowable" ruleset="Error Prone" package="org.apache.geronimo.console.internaldb" class="RunSQLHelper" method="runSQL" externalInfoUrl="https://pmd.github.io/pmd-6.44.0/pmd_rules_java_errorprone.html#avoidcatchingthrowable" priority="3">
A catch statement should never catch throwable since it includes errors.
</violation>
</file>
</pmd>
